### ISSUE TITLE
force option not working as documented

### ISSUE THREAD

**Issue by memeplex** [vitalif/grive2#51]
```
  -f [ --force ]        Force grive to always download a file from Google Drive
                        instead of uploading it.
```

but when I locally change a file that's already in google drive, grive2 uploads it despite the -f setting.

**Comment by memeplex** [comment]
I'm not 100% sure I'm correctly interpreting the option documentation.

Anyway, my use case is that I want to start with a "checkout" or "clone" of the server. For example, if I'm sync'ing my dotfiles to a new computer I don't want the `.bashrc` installed by default to override my custom `.bashrc` in drive at any rate!

**Comment by memeplex** [comment]
Seems like 

```
		// if remote is modified
		if ( remote.MTime() > m_mtime )
		{
			Log( "file %1% is changed in remote", path, log::verbose ) ;
			m_state = remote_changed ;
		}
```

in Resource.cc is completely ignoring `last_change`, which is the mechanism by which force is implemented:

```
	if ( force )
	{
		m_last_change = DateTime() ;
		m_last_sync = DateTime::Now() ;
	}
```

**Comment by memeplex** [comment]
This could be a fix:

```
		// if remote is modified
		if ( remote.MTime() > last_change || remote.MTime() > m_mtime )
		{
			Log( "file %1% is changed in remote", path, log::verbose ) ;
			m_state = remote_changed ;
		}
```

but I think it would be more readable to explicity check the force flag instead of last_change.

**Comment by vitalif** [comment]
You're right, that's another bug introduced while trying to fix something...

The correct fix I think would be to change m_last_change to m_mtime in if(force)

**Comment by vitalif** [comment]
See the master branch. I done a major rework today, adding support for local checksum and mtime cache ("index"). Force option should also work now :)
